sklearn/cluster/_kmeans.py

```diff
@@ -275,12 +275,49 @@ class KMeansCythonEngine:
     TODO: see URL for more details.
     """
 
+    @staticmethod
+    def convert_to_sklearn_types(name, value):
+        """Convert estimator attributes to scikit-learn types.
+
+        Users can configure whether estimator attributes should be stored
+        using engine native types or scikit-learn types. This function is
+        used to convert attributes from engine to scikit-learn native types.
+
+        Scikit-learn native types are ndarrays and basic Python types. There
+        is no need to convert these.
+
+        Parameters
+        ----------
+        name : str
+            Name of the attribute being converted.
+
+        value
+            Value of the attribute being converted.
+
+        Returns
+        --------
+        converted
+            Attribute value converted to a scikit-learn native type.
+        """
+        # XXX Maybe a bit useless as it should never get called, but it
+        # does demonstrate the API
+        return value
+
     def __init__(self, estimator):
         self.estimator = estimator
 
-    def accepts(self, X, y=None, sample_weight=None):
-        # The default engine accepts everything
-        return True
+    def validate(self, X, y=None, sample_weight=None):
+        """Validate input and hyper-parameters.
+
+        Determine if the engine can handle the hyper-parameter of the estimator
+        as well as the input data. Input data may be converted and then validated,
+
+        Return the acceptance decision and a dictionary containing the input data.
+
+        Should fail as quickly as possible.
+        """
+        # The default engine accepts everything and does not convert inputs
+        return True, {"X": X, "y": y, "sample_weight": sample_weight}
```

Comment:
```suggestion
        # The default engine accepts everything and does not convert inputs
        accept_inputs = True
        return accept_inputs, {"X": X, "y": y, "sample_weight": sample_weight}
```

Reply:
It's not just about accepting the inputs but the inputs in conjunction with the estimator hyper-parameters.

I think it's ok for an engine to accept to be enabled only for some hyper-parameters, even if the inputs are all valid (e.g. depending of the `estimator.algorithm` parameter for `KMeans`, lloyd vs elkan).

So I would rather keep the short `accept` variable name.

Reply:
Then:
```suggestion
        # The default engine accepts everything and does not convert inputs
        accept = True
        return accept, {"X": X, "y": y, "sample_weight": sample_weight}
```